### PR TITLE
feat(priority): Update group serializer to return priority

### LINKED ISSUE #63266: Update GET issue endpoint to return priority

### PR DESCRIPTION
**Pull request by 17hogeju**
Closes https://github.com/getsentry/sentry/issues/63266

Serializer updated to return priority. The serialized value is a string (high/ medium / low) and not the raw number value.

### CODE REVIEW

File: src/sentry/api/serializers/models/group.py [hunk 1]
```diff
@@ -392,6 +395,16 @@ def _collapse(self, key) -> bool:
             return False
         return key in self.collapse
 
+    def _get_priority(self, obj: Group) -> str:
+        priority = obj.priority
+        if priority == PriorityLevel.HIGH:
+            priority_label = "high"
+        elif priority == PriorityLevel.MEDIUM:
+            priority_label = "medium"
+        else:
```
Comment (17hogeju) [inline src/sentry/api/serializers/models/group.py:404]: Should `PriorityLevel.LOW` be a default? Do we need to worry about a case where Group Priority is None?

Comment (snigdhas) [reply src/sentry/api/serializers/models/group.py:404]: The default priority should be `PriorityLevel.MEDIUM`, per https://www.notion.so/sentry/Tech-Spec-Priority-in-Issue-Details-Stream-ac9ecbbc5cfd4baaae39ecf32f8e74ce?pvs=4#ae9b3ffebd4a4859b940f5b50e2ddba3

Comment (snigdhas) [reply src/sentry/api/serializers/models/group.py:404]: We should add a feature flag check here, and support none until the migration is complete and we're actively setting priority on each new issue

Comment (17hogeju) [reply src/sentry/api/serializers/models/group.py:404]: Ah yes, thanks for the reminder!

File: src/sentry/api/serializers/models/group.py [hunk 2]
```diff
@@ -326,6 +327,7 @@ def serialize(
         self, obj: Group, attrs: MutableMapping[str, Any], user: Any, **kwargs: Any
     ) -> BaseGroupSerializerResponse:
         status_details, status_label = self._get_status(attrs, obj)
+        priority_label = self._get_priority(obj)
```
Comment (snigdhas) [inline src/sentry/api/serializers/models/group.py:330]:
```suggestion
        priority_label = obj.priority.name if obj.priority is not None else None
```

Comment (17hogeju) [reply src/sentry/api/serializers/models/group.py:330]: Since priority is coming straight from the group model, it does not have a 'name' property

Comment (17hogeju) [reply src/sentry/api/serializers/models/group.py:330]: Also, do we want to ever have the label as type 'None'?

Comment (snigdhas) [reply src/sentry/api/serializers/models/group.py]: where is this getting used? if we're serializing for the FE, we can skip the field if the flag is off or the value is None.

Comment (snigdhas) [reply src/sentry/api/serializers/models/group.py]: I'm adding this map in another PR - if it's useful feel free to copy it in and I'll deal with the merge conflict later lol:

```
PRIORITY_LEVEL_TO_STR: dict[int, str] = {
    PriorityLevel.LOW: "low",
    PriorityLevel.MEDIUM: "medium",
    PriorityLevel.HIGH: "high",
}
```

File: tests/snuba/api/serializers/test_group.py [hunk 3]
```diff
@@ -45,6 +47,35 @@ def test_permalink_outside_org(self):
...
+        outside_user = self.create_user()
+        group = self.create_group()
+        result = serialize(group, outside_user, serializer=GroupSerializerSnuba())
+        assert "priority" not in result
+
+    @with_feature("projects:issue-priority")
+    def test_priority_high(self):
+        outside_user = self.create_user()
+        group = self.create_group()
+        group.priority = PriorityLevel.HIGH
```
Comment (snigdhas) [inline tests/snuba/api/serializers/test_group.py:60]: here and below: 

```suggestion
        group = self.create_group(priority=PriorityLevel.HIGH)
```

File: src/sentry/api/serializers/models/group.py (hunk 2 above)
Comment (17hogeju) [reply src/sentry/api/serializers/models/group.py]: @snigdhas On the GroupDetailsEndpoint(`GET /issues/<issue_id/`) when the data is serialized using GroupSerializerSnuba which has GroupSerializerBase as a parent.

**Review by snigdhas — APPROVED** [review]

**Review by malwilley — APPROVED** [review]

**Pull request merged**